### PR TITLE
fixed broken story userTeamsTable

### LINKED ISSUE #301: Fix broken story at http://localhost:6006/?path=/story/tilde-userteamstable--primary

### PR DESCRIPTION
**Pull request by kaleem99**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kaleem99** [comment]
image

**Review by Vuyisanani — APPROVED** [review]

File: frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ const TeamsTable = ({ teams, authUser }) => {
                 <TableCell>{team.name}</TableCell>
                 <TableCell>
                   {(authUser.isSuperuser ||
-                    authUser.permissions.teams[team.id]) && (
+                    authUser.teamMemberships[team]) && (
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js]: Please don't fix the story in such a way as to break the actual frontend. 

We need the story to work AND we need Tilde to not break.

If you look at the redux store in the running application you will see that the authUser object looks lvery different from what you put there.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Please fix the story without breaking anything else. It's important not to introduce new bugs.

**Review by Sbonelo01 — APPROVED** [review]

**Review by Sbonelo01 — CHANGES_REQUESTED** [review]
Wrong PR approved. please dismiss the review and push the necessary changes

**Review by Sbonelo01 — APPROVED** [review]

File: frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js [hunk 2]
```diff
@@ -39,10 +39,11 @@ const TeamsTable = ({ teams, authUser }) => {
           {teams ? (
             Object.values(teams).map((team) => (
               <TableRow key={Object.values(teams).indexOf(team)}>
+                  {console.log(authUser)}
```
Comment (FaithMo) [inline frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js]: Please remove printing statement

**Review by FaithMo — CHANGES_REQUESTED** [review]

**Review by FaithMo — APPROVED** [review]

File: frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js [hunk 3]
```diff
@@ -41,8 +41,8 @@ const TeamsTable = ({ teams, authUser }) => {
               <TableRow key={Object.values(teams).indexOf(team)}>
                 <TableCell>{team.name}</TableCell>
                 <TableCell>
-                  {(authUser.isSuperuser ||
-                    authUser.permissions.teams[team.id]) && (
+                {(authUser.isSuperuser ||
+                    authUser.permissions) && (
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js]: authUser.permissions.teams[team.id] basically says: Does the authenticated have a permission over the current team.

If the authenticated user does not have permission then they should not see the view button. 

Remember that this code works in production. Just the story is broken. Please dont break production code in order to make the story work. Just fix the story.

File: frontend/src/stories/TeamsTable.stories.js [hunk 4]
```diff
@@ -12,4 +12,5 @@ const Template = args => <TeamsTable {...args} />
 export const Primary = Template.bind({});
 Primary.args = {
   teams: user.teamMemberships,
+  authUser: user
```
Comment (sheenarbw) [inline frontend/src/stories/TeamsTable.stories.js]: What's the difference between  "./fixtures/user.json" and  "./fixtures/authUser.json"?

Please make sure you pass the right object in.

The authUser is the user that is currently authenticated.
The user if the person we are looking at. 

So if Kaleem logs in and looks at Babalwa's info, then the authuser is Kaleem and the the user is Babalwa.

The data structures are different.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
You shouldn't be making changes to TeamsTable/Presentations.js. 

You just need to make the story work. The component already does the right thing. Just the story is broken.

**Review by babalwa-mbolekwa — APPROVED** [review]
Hey Kaleem. Well done. There are also some other broken stories but they are not mentioned in the ticket maybe there will be other tickets created for them:
- DueTime AgileCard story and `ProjectDetails` story

**Review by FaithMo — APPROVED** [review]

File: frontend/src/stories/TeamsTable.stories.js [hunk 5]
```diff
@@ -11,5 +11,6 @@ const Template = args => <TeamsTable {...args} />
 
 export const Primary = Template.bind({});
 Primary.args = {
-  teams: user.teamMemberships,
+  teams: authUser.teamMemberships,
```
Comment (sheenarbw) [inline frontend/src/stories/TeamsTable.stories.js]: The authuser and the user are not necessarily the exact same person. You Kaleem, might be looking at Babalwa's board. That would mean that the authuser is you, Kaleem. And the user is Babalwa.

File: frontend/src/stories/TeamsTable.stories.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import TeamsTable from "../components/regions/UserDetails/TeamsTable/Presentation";
-import user from "./fixtures/user.json";
```
Comment (sheenarbw) [inline frontend/src/stories/TeamsTable.stories.js:3]: please make use of the user AND the authUser. They have different purposes. Both are relevent

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Please make sure that this works if you are not looking at your own team memberships. Mostly when making use of this component you would be looking at other people's team memberships.

Please make sure you know the difference between the authenticated user and any other user. 

If I am logged in then I am the authenticated user.
When I am authenticated then I can look at any users team memberships.  I probabaly dont care too much about looking at my own team memberships.  

Please use the right data for the right purpose

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**